external-crates/move/crates/move-transactional-test-runner/src/framework.rs

```diff
@@ -756,23 +758,65 @@ where
             None
         }
     };
+
     let (mut adapter, result_opt) =
         Adapter::init(default_syntax, fully_compiled_program_opt, init_opt, path).await;
+
     if let Some(result) = result_opt {
         if let Err(e) = writeln!(output, "\ninit:\n{}", result) {
-            // TODO: if this fails, it masks the actual error, need better error handling
-            // in case cleanup_resources() fails
+            // TODO: if this fails, it masks the actual error, need better error handling in case
+            // cleanup_resources() fails
             adapter.cleanup_resources().await?;
             return Err(Box::new(e));
         }
     }
+
+    Ok((output, adapter))
+}
+
+/// Consumes the adapter to run tasks from path.
+pub async fn run_tasks_with_adapter<'a, Adapter>(
+    path: &Path,
+    mut adapter: Adapter,
+    mut output: String,
+) -> Result<()>
+where
+    Adapter: MoveTestAdapter<'a>,
+    Adapter::ExtraInitArgs: Debug,
+    Adapter::ExtraPublishArgs: Debug,
+    Adapter::ExtraValueArgs: Debug,
+    Adapter::ExtraRunArgs: Debug,
+    Adapter::Subcommand: Debug,
+{
+    let mut tasks = taskify::<
+        TaskCommand<
+            Adapter::ExtraInitArgs,
+            Adapter::ExtraPublishArgs,
+            Adapter::ExtraValueArgs,
+            Adapter::ExtraRunArgs,
+            Adapter::Subcommand,
+        >,
+    >(path)?
+    .into_iter()
+    .collect::<VecDeque<_>>();
+    assert!(!tasks.is_empty());
+
+    // Pop off init command if present
```

Comment:
nit: comment to explain that it was already handled before this function was called.